<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="gpsimu_en" default="false" />
  <arg name="urg_en" default="false" />
  <arg name="ladybug_en" default="false" />
  <arg name="velodyne_en" default="false" />
  <arg name="flea2_en" default="false" />
  <arg name="canbus_en" default="false" />
  
  <arg name="gpsport" default="/dev/ttyS1" />
  <arg name="gpsbaud" default="115200" />
  <arg name="gpsverbose" default="true" />
  <arg name="gpssavelog" default="true" />
  <arg name="gpsupdaterate" default="100" />
  <arg name="gpsrate" default="0.01" />

  <arg name="urgip" default="192.168.0.10" />
  <arg name="urgport" default="/dev/ttyACM0" />
  <arg name="urgbaud" default="115200" />
  <arg name="urgframeid" default="laser" />


  <!--gpsimu node  -->
  <group if="$(arg gpsimu_en)" >    
    <node pkg="poss_jeep" type="xwgpsimu.py" name="Xwgpsimu" output="screen">
      <param name="port" value="$(arg gpsport)" />
      <param name="baud" value="$(arg gpsbaud)" />
      <param name="verbose" value="$(arg gpsverbose)" />
      <param name="savelog" value="$(arg gpssavelog)" />
      <param name="gpsrate" value="$(arg gpsrate)" />
      <param name="updaterate" value="$(arg gpsupdaterate)" />
    </node>
  </group>

  <!--urg node  -->
  <group if="$(arg urg_en)" >
    <node name="urg_node" pkg="urg_node" type="urg_node" output="screen">
      <param name="ip_address" value="$(arg urgip)"/>
      <param name="serial_port" value="$(arg urgport)"/>
      <param name="serial_baud" value="$(arg urgbaud)"/>
      <param name="frame_id" value="$(arg urgframeid)"/>
      <param name="calibrate_time" value="true"/>
      <param name="publish_intensity" value="false"/>
      <param name="publish_multiecho" value="false"/>
      <param name="angle_min" value="-1.5707963"/>
      <param name="angle_max" value="1.5707963"/>
    </node>
  </group>

  <!--official ladybug recorder -->
  <group if="$(arg ladybug_en)">
    <node pkg="poss_jeep" type="ladybugrec.sh" name="ladybugrecorder" output="screen">
    </node>
  </group>

  <!--official velodyne -->
  <group if="$(arg velodyne_en)">
    <node pkg="poss_jeep" type="vdump.sh" name="velodyne_recoder" output="screen" args="eno1">
    </node>
  </group>


</launch>
